chips/nrf52/src/uart.rs

```diff
@@ -121,6 +120,10 @@ impl UART {
         let regs = unsafe { &*self.regs };
         let tx = regs.event_endtx.get() != 0;
 
+        if regs.event_error.get() == 1 {
+            panic!("uart error event\r\n");
```

Comment:
In general I think the right thing to do is abort the transaction and propagate the error up (through the client callback's error argument). Though i don't exactly know what error conditions are possible on the NRF's uart